### PR TITLE
Implementing Latin Hypercube Sampling

### PR DESCRIPTION
**Pull request by elacx**
We implement LHS for tabular data. This allows for faster convergence of the explanations.

### CODE REVIEW

File: lime/lime_tabular.py [hunk 1]
```diff
@@ -189,6 +191,7 @@ def __init__(self,
         self.categorical_names = categorical_names or {}
         self.sample_around_instance = sample_around_instance
         self.training_data_stats = training_data_stats
+        self.model = None #to access model after the fact
```
Comment (marcotcr) [inline lime/lime_tabular.py:194]: Do we need this?

Comment (elacx) [reply lime/lime_tabular.py:194]: I can remove that, I was using this previously to investigate things as I was working on understanding the LIME code.

**Comment by elacx** [comment]
It looks like the Python 3.6 build passed but the 3.7 one did not. There is one failure occurring in the test_lime_explainer_entropy_discretizer (which I did not alter). A few of the previous builds passed for both Python 3.6 and 3.7.

I will be adding the notebook to doc/notebooks within the week.

Thanks!

**Comment by elacx** [comment]
Looks like the build has passed and I have added the lhs notebook. I think once its accepted the link in the read me file to the lhs example will need to be updated but that is it (maybe on your end because I don't know what the link for the lhs file will look like after the merge). Please let me know if you need anything else on my end, thanks!

Screen Shot 2020-07-28 at 9 51 32 AM

**Pull request merged**